### PR TITLE
Add support for legacy MONITORING| line in forwarder

### PR DESCRIPTION
**Pull request by DarcyRaynerDD**
### What does this PR do?

Adds support for the legacy "MONITORING|" metric format.

### Motivation

To make customer migration from legacy MONITORING| format easier.

### Types of changes

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)
- [ ] This PR passes the unit tests 
- [ ] This PR passes the installation tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: aws/logs_monitoring/lambda_function.py [hunk 1]
```diff
@@ -56,6 +56,7 @@
     DD_ADDITIONAL_TARGET_LAMBDAS,
     DD_USE_VPC,
     DD_MAX_WORKERS,
+    DD_LEGACY_MONITORING_LINE,
```
Comment (tianchu) [inline aws/logs_monitoring/lambda_function.py]: How about `DD_PARSE_LEGACY_MONITORING_LOGS`?

File: aws/logs_monitoring/lambda_function.py [hunk 2]
```diff
@@ -635,7 +636,13 @@ def extract_trace_payload(event):
 def extract_metric(event):
     """Extract metric from an event if possible"""
     try:
-        metric = json.loads(event["message"])
+        message = event["message"]
```
Comment (tianchu) [inline aws/logs_monitoring/lambda_function.py:639]: Perhaps add a short comment?

File: aws/logs_monitoring/lambda_function.py [hunk 3]
```diff
@@ -655,6 +662,29 @@ def extract_metric(event):
...
         return None
 
+def extract_legacy_metric(message):
+    message = message.strip()
+    values = message.split("|")
+    if len(values) != 6:
+        return None
+    _, ts, value, type, name, tags = values
+    
+    timestamp = int(float(ts))  # float to catch decimals
```
Comment (tianchu) [inline aws/logs_monitoring/lambda_function.py:676]: Our docs says it should always be in seconds https://docs.datadoghq.com/serverless/custom_metrics?tab=python#deprecated-using-cloudwatch-logs?

File: aws/logs_monitoring/settings.py [hunk 4]
```diff
@@ -226,3 +226,5 @@ def __init__(self, name, pattern, placeholder):
 
 DD_TAGS_CACHE_TTL_SECONDS = int(get_env_var("DD_TAGS_CACHE_TTL_SECONDS", default=300))
 DD_S3_CACHE_LOCK_TTL_SECONDS = 60
+
+DD_LEGACY_MONITORING_LINE = get_env_var("DD_LEGACY_MONITORING_LINE", "false", boolean=True)
```
Comment (tianchu) [inline aws/logs_monitoring/settings.py]: Can you add it as an Advanced parameter in the CF template?

**Review by tianchu — APPROVED** [review]
Looks good, except for a few minor comments.

File: aws/logs_monitoring/lambda_function.py (hunk 3 above)
Comment (DarcyRaynerDD) [reply aws/logs_monitoring/lambda_function.py:676]: I adapted the logic from the crawler, which had this edge case as well.

File: aws/logs_monitoring/lambda_function.py (hunk 1 above)
Comment (DarcyRaynerDD) [reply aws/logs_monitoring/lambda_function.py]: Sure

File: aws/logs_monitoring/lambda_function.py [hunk 5]
```diff
@@ -635,7 +636,13 @@ def extract_trace_payload(event):
 def extract_metric(event):
     """Extract metric from an event if possible"""
     try:
-        metric = json.loads(event["message"])
+        message = event["message"]
+        if DD_LEGACY_MONITORING_LINE and message.strip().startswith("MONITORING|"):
+            metric = extract_legacy_metric(message)
+            if metric:
+                return metric
```
Comment (tianchu) [inline aws/logs_monitoring/lambda_function.py]: perhaps we should also add `metric["t"] += event[DD_CUSTOM_TAGS].split(",")` before return?

File: aws/logs_monitoring/template.yaml [hunk 6]
```diff
@@ -154,6 +154,14 @@ Parameters:
       - true
       - false
     Description: Set to true to deploy the Forwarder to a VPC and send logs, metrics, and traces via a proxy. When set to true, must also set VPCSecurityGroupIds and VPCSubnetIds.
+  DdParseLegacyMonitoringLogs:
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+    Description: Set to true to enable the support for converting legacy MONITORING| formatted lines to distribution metrics.
```
Comment (tianchu) [inline aws/logs_monitoring/template.yaml]: Let's also mention "`.dist` will be prepended to metric names to differentiate from the non-distribution ones.

**Review by tianchu — APPROVED** [review]

**Pull request merged**